crates/e2e/tests/e2e/cow_amm.rs

```diff
@@ -595,3 +596,685 @@ async fn cow_amm_driver_support(web3: Web3) {
     .await
     .unwrap();
 }
+
+#[tokio::test]
+#[ignore]
+async fn local_node_cow_amm_opposite_direction() {
+    run_test(cow_amm_opposite_direction).await;
+}
+
+/// Tests that only CoW AMM liquidity can be used to fulfill the order.
+async fn cow_amm_opposite_direction(web3: Web3) {
+    let mut onchain = OnchainComponents::deploy(web3.clone()).await;
+
+    let [solver] = onchain.make_solvers(to_wei(100)).await;
+    let [bob, cow_amm_owner] = onchain.make_accounts(to_wei(1000)).await;
+
+    let [dai] = onchain
+        .deploy_tokens_with_weth_uni_v2_pools(to_wei(300_000), to_wei(100))
+        .await;
+
+    // No need to fund the buffers since we're testing the CoW AMM directly filling
+    // the user order.
+
+    // Set up the CoW AMM as before
+    let oracle = contracts::CowAmmUniswapV2PriceOracle::builder(&web3)
+        .deploy()
+        .await
+        .unwrap();
+
+    let cow_amm_factory = contracts::CowAmmConstantProductFactory::builder(
+        &web3,
+        onchain.contracts().gp_settlement.address(),
+    )
+    .deploy()
+    .await
+    .unwrap();
+
+    // Fund the CoW AMM owner with DAI and WETH and approve the factory to transfer
+    // them
+    dai.mint(cow_amm_owner.address(), to_wei(2_000)).await;
+    tx!(
+        cow_amm_owner.account(),
+        dai.approve(cow_amm_factory.address(), to_wei(2_000))
+    );
+
+    tx_value!(
+        cow_amm_owner.account(),
+        to_wei(1),
+        onchain.contracts().weth.deposit()
+    );
+    tx!(
+        cow_amm_owner.account(),
+        onchain
+            .contracts()
+            .weth
+            .approve(cow_amm_factory.address(), to_wei(1))
+    );
+
+    tx_value!(
+        solver.account(),
+        to_wei(1),
+        onchain.contracts().weth.deposit()
+    );
+
+    // Fund the settlement contract with WETH so it can pay out the user order.
```

Comment:
Then, I assume there is still something that needs to be corrected since the simulation fails without this funding for both quote and settlement e2e tests.